components/edit-collective/CreateVirtualCardModal.js

```diff
@@ -0,0 +1,285 @@
+import { P } from '../Text';
+import { TOAST_TYPE, useToasts } from '../ToastProvider';
+
+const createVirtualCardMutation = gqlV2/* GraphQL */ `
+  mutation createVirtualCard(
+    $name: String!
+    $monthlyLimit: AmountInput!
+    $account: AccountReferenceInput!
+    $assignee: AccountReferenceInput!
+  ) {
+    createVirtualCard(name: $name, monthlyLimit: $monthlyLimit, account: $account, assignee: $assignee) {
+      id
+      name
+      last4
+      data
+    }
+  }
+`;
+
+// TODO : refactor this mutation
+const collectiveMembersQuery = gqlV2/* GraphQL */ `
+  query CollectiveMembers($slug: String!) {
+    account(slug: $slug) {
+      id
+      members(role: ADMIN) {
+        nodes {
+          account {
+            id
+            name
+            imageUrl
+            slug
+          }
+        }
+      }
+    }
+  }
+`;
+
+const throttledCall = debounce((searchFunc, variables) => {
+  return searchFunc({ variables });
+}, 750);
+
+const initialValues = {
+  collective: undefined,
+  assignee: undefined,
+  cardName: undefined,
+  monthlyLimit: undefined,
+};
+
+const CreateVirtualCardModal = ({ host, onSuccess, onClose, ...modalProps }) => {
+  const { addToast } = useToasts();
+
+  const [createVirtualCard, { loading: isBusy }] = useMutation(createVirtualCardMutation, {
+    context: API_V2_CONTEXT,
+  });
+  const [getCollectiveUsers, { loading: isLoadingUsers, data: users }] = useLazyQuery(collectiveMembersQuery, {
+    context: API_V2_CONTEXT,
+  });
+
+  const formik = useFormik({
+    initialValues: { ...initialValues },
+    async onSubmit(values) {
+      const { collective, assignee, cardName, monthlyLimit } = values;
+
+      try {
+        await createVirtualCard({
+          variables: {
+            assignee: { id: assignee.id },
+            account: typeof collective.id === 'string' ? { id: collective.id } : { legacyId: collective.id },
+            name: cardName,
+            monthlyLimit: { currency: 'USD', valueInCents: monthlyLimit, value: monthlyLimit / 100 },
+          },
+        });
+      } catch (e) {
+        addToast({
+          type: TOAST_TYPE.ERROR,
+          message: (
+            <FormattedMessage
+              defaultMessage="Error creating virtual card: {error}"
+              values={{
+                error: e.message,
+              }}
+            />
+          ),
+        });
+        return;
+      }
+      onSuccess?.();
+      handleClose();
+    },
+    validate(values) {
+      const errors = {};
+      if (!values.collective) {
+        errors.collective = 'Required';
+      }
+      if (!values.assignee) {
+        errors.assignee = 'Required';
+      }
+      if (!values.cardName) {
+        errors.cardName = 'Required';
+      }
+      if (!values.monthlyLimit) {
+        errors.monthlyLimit = 'Required';
+      }
+      if (values.monthlyLimit > 100000) {
+        errors.monthlyLimit = 'MonthlyLimit should not exceed 1000$';
```

Comment:
To not forget, this is actually more important to check this on the API side than here.

We should look at better localization at some point.

Reply:
Sorry, I double checked and we properly check the monthlyLimit on API side.

Reply:
It is also checked on the api side